src/pages/stacks-blockchain/stacking.md

```diff
@@ -1,50 +1,54 @@
+## Token holder eligibility
 
-- must determine the set of addresses that miners may validly transfer funds to. Consensus of this requires a "rewarding prepare" phase in which an anchor block and reward set are decided on
-- Progression in Stacking consensus happens over reward cycles (with a fixed length). In each reward cycle, a set of Bitcoin addresses are iterated over, such that each Bitcoin address in the set of reward addresses has exactly one Bitcoin block in which miners will transfer funds to the reward address.
-- If a miner is building off a descendant of the anchor block, the miner must send commitment funds to 5 addresses from the reward set
+Stacks token holders do not automatically receive Stacking rewards. Instead, they must:
 
-### Integrating stacking
+- Commit to participation before a reward cycle begins
+- Hold ~94.000 Stacks tokens
+- Lock up Stacks tokens for a specified period
+- Set a Bitcoin address to receive rewards
 
-Ready to integrate stacking in your wallet or exchange?
+Token holders will have to use a wallet or exchange that supports participation in Stacking.
 
 [@page-reference | inline]
 | /stacks-blockchain/integrate-stacking
+
+## Stacking consensus algorithm
+
+Stacking is a built-in capability of PoX and is realized through a set of actions on the Stacks 2.0 network. The full implementation details can be found in [SIP-007](https://github.com/blockstack/stacks-blockchain/blob/develop/sip/sip-007-stacking-consensus.md). Below is a summary of the most relevant actions of the algorithm.
+
+-> Fun fact: The Stacking consensus algorithm is implemented as a smart contract, using [Clarity](/smart-contracts/overview).
+
+- Progression in Stacking consensus happens over reward cycles (with a fixed length). In each reward cycle, a set of Bitcoin addresses are iterated over
+- A reward cycle consists of two phases: prepare and reward
+- During the prepare phase, miners decide on an anchor block and a reward set. Mining any descendant forks of the anchor block requires transferring mining funds to the appropriate reward addresses. The reward set is the set of Bitcoin addresses which will receive funds in the reward cycle
+- Miners register as leader candidates for a future election by sending a key transaction that both burns cryptocurrency (proof-of-burn) and spends energy (proof-of-work). The transaction also registers the leader's preferred chain tip (must be a descendant of the anchor block) and commitment of funds to 5 addresses from the reward set
+- Token holders register for the next rewards cycle by broadcasting a signed message that locks up associated Stacks tokens for a protocol-specified lockup period, specifies a Bitcoin address to receive the funds, and votes on a Stacks chain tip
+- The leader that commits to the winning chain tip and the peers who also burn for that leader collectively share the reward, proportional to how much each one burned
```

Comment:
multiple leaders can commit to the same chain tip (hopefully, they all do, or this means a fork), it is the leader that wins the sortition, and whoever supported him that will get the reward.
That leader either transfers (or burns) btc, depending on several factors.

Reply:
thanks!